subnet/deployment-generator/src/config_gen.js

```diff
@@ -1,176 +1,232 @@
-const crypto = require('crypto')
-const net = require('net')
-const dotenv = require('dotenv')
-const ethers = require('ethers')
-dotenv.config({ path: `${__dirname}/gen.env` })
+const crypto = require("crypto");
+const net = require("net");
+const dotenv = require("dotenv");
+const ethers = require("ethers");
+dotenv.config({ path: `${__dirname}/gen.env` });
 // console.log(__dirname)
 
 const config = {
-  deployment_path: (process.env.CONFIG_PATH || ''),
-  num_machines:  parseInt(process.env.NUM_MACHINE),
-  num_subnet:         parseInt(process.env.NUM_SUBNET),
-  ip_1:               (process.env.MAIN_IP                || ''),
-  public_ip:          (process.env.PUBLIC_IP || process.env.MAIN_IP),
-  network_name:       (process.env.NETWORK_NAME),
-  network_id:         parseInt(process.env.NETWORK_ID     || Math.floor(Math.random() * (65536 - 1) + 1)),
-  secret_string:      (process.env.SERVICES_SECRET        || crypto.randomBytes(10).toString('hex')),
-  relayer_mode:       (process.env.RELAYER_MODE           || 'full'), //full or lite //in upgradable csc both are deployed
-  docker_image_name:  (process.env.IMAGE_NAME             || 'xinfinorg/subnet-generator:latest'),
-  operating_system:   (process.env.OS                     || 'linux'),
+  deployment_path: process.env.CONFIG_PATH || "",
+  num_machines: parseInt(process.env.NUM_MACHINE),
+  num_subnet: parseInt(process.env.NUM_SUBNET),
+  ip_1: process.env.MAIN_IP || "",
+  public_ip: process.env.PUBLIC_IP || process.env.MAIN_IP,
+  network_name: process.env.NETWORK_NAME,
+  network_id: parseInt(
+    process.env.NETWORK_ID || Math.floor(Math.random() * (65536 - 1) + 1)
+  ),
+  secret_string:
+    process.env.SERVICES_SECRET || crypto.randomBytes(10).toString("hex"),
+  relayer_mode: process.env.RELAYER_MODE || "full",
+  docker_image_name:
+    process.env.IMAGE_NAME || "xinfinorg/subnet-generator:latest",
+  operating_system: process.env.OS || "linux",
   version: {
-    subnet:   (process.env.VERSION_SUBNET   || 'v0.2.2'),   
-    bootnode: (process.env.VERSION_BOOTNODE || 'v0.2.2'),
-    // observer: (process.env.VERSION_OBSERVER || 'latest'),
-    relayer:  (process.env.VERSION_RELAYER  || 'v0.2.2'),
-    stats:    (process.env.VERSION_STATS    || 'v0.1.8'),   
-    frontend: (process.env.VERSION_FRONTEND || 'v0.1.10'),   
-    csc:      (process.env.VERSION_CSC      || 'v0.1.1'),
-    // zero:     (process.env.VERSION_ZERO     || 'v0.1.1')
+    subnet: process.env.VERSION_SUBNET || "feature-v1-release",
+    bootnode: process.env.VERSION_BOOTNODE || "feature-v1-release",
+    relayer: process.env.VERSION_RELAYER || "feature-v1-release",
+    stats: process.env.VERSION_STATS || "feature-v1-release",
+    frontend: process.env.VERSION_FRONTEND || "feature-v1-release",
+    csc: process.env.VERSION_CSC || "feature-v0.2.1",
+    zero: process.env.VERSION_ZERO || "v0.1.0",
```

Comment:
I think it's future, you can put default as latest, once you are ready for the release, update the latest image.

Reply:
I don't like using latest here, since not all versions are guaranteed to be compatible. We also run into problems like whether the image is really 'latest' or the user has not run 'docker pull'. Also for troubleshooting it's hard to match 'latest' to a known version.

Reply:
make sense